song/song_server.py

```diff
@@ -62,19 +64,22 @@ def __init__(self, osculator_client, display_client, machine, beat_manager):
             range(len(self.song_machine.parser.states)
                   ))}
 
+        self.osculator_client.send_message(settings.SONG_ADVANCE_ADDRESS, (1, 1.0))
+        self.osculator_client.send_message(settings.SONG_ADVANCE_ADDRESS, (1, 0.0))
+        self.tonality = Tonality(self.song_machine.parser.categories)
         self._send_init_to_display()
 
-        self.osculator_client.send_message(settings.SONG_ADVANCE_ADDRESS, (9, 1.0))
-        self.osculator_client.send_message(settings.SONG_ADVANCE_ADDRESS, (9, 0.0))
-        self._send_partinfo_to_display()
-
     def interpreter_handler(self, _, content):
         if self.song_machine.is_locked():
             return
 
         osc_map = pickle.loads(content)
-        print('SongerServer. receiving: ', osc_map)
-        self._send_level(osc_map['level'])
+        self.send_FX()
+
+        self.tonality.update_tonality(osc_map['cat'])
```

Comment:
This might have changed by me (while merging back master), but shouldn't the order be reversed, i.e. FIRST `update_tonality` and THEN `send_FX()`?